### PR TITLE
Fixed editor gui

### PR DESCRIPTION
**Pull request by Sajdkick**
Made the GUI stick to the sides of the window and now you can only open one resource at a time.

### CODE REVIEW

**Comment by Chainsawkitten** [comment]
It's unnecessary to keep a map of editors if only one can be open anyhow. This makes the code needlessly complex.

File: src/Editor/Editor.cpp [hunk 1]
```diff
@@ -19,8 +20,14 @@ Editor::Editor() {
 void Editor::Show() {
     bool play = false;
     
+    ImVec2 size(MainWindow::GetInstance()->GetSize().x, MainWindow::GetInstance()->GetSize().y);
+    ImVec2 menu_size;
```
Comment (Chainsawkitten) [inline src/Editor/Editor.cpp]: Variable name doesn't follow naming conventions.

File: src/Editor/GUI/Editors/SceneEditor.cpp [hunk 2]
```diff
@@ -8,19 +8,20 @@
 using namespace GUI;
 
 void SceneEditor::Show() {
-    if (ImGui::Begin(("Scene: " + *scene + "###Scene").c_str(), &visible)) {
+    if (ImGui::Begin(("Scene: " + std::to_string(sceneIndex) + "###Scene").c_str(), &visible)) {
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.cpp]: Should show actual scene name instead.

File: src/Editor/GUI/Editors/SceneEditor.cpp [hunk 3]
```diff
@@ -60,7 +58,6 @@ void SceneEditor::SetVisible(bool visible) {
 }
 
 void SceneEditor::Save() const {
-    if (scene != nullptr) {
-        Hymn().world.Save(Hymn().GetPath() + FileSystem::DELIMITER + "Scenes" + FileSystem::DELIMITER + *scene + ".json");
-    }
+	if(sceneIndex < Hymn().scenes.size())
+		Hymn().world.Save(Hymn().GetPath() + FileSystem::DELIMITER + "Scenes" + FileSystem::DELIMITER + std::to_string(sceneIndex) + ".json");
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.cpp]: Filename should be the scene name, not its index.

File: src/Editor/GUI/ResourceList.hpp [hunk 4]
```diff
@@ -48,9 +48,11 @@ namespace GUI {
         private:
             bool visible = false;
             
-            SceneEditor sceneEditor;
-            std::map<Geometry::Model*, ModelEditor> modelEditors;
-            std::map<Texture2D*, TextureEditor> textureEditors;
-            std::map<Audio::SoundBuffer*, SoundEditor> soundEditors;
+			SceneEditor sceneEditor;
```
Comment (Chainsawkitten) [inline src/Editor/GUI/ResourceList.hpp]: Tabs

File: src/Editor/GUI/ResourceList.cpp [hunk 5]
```diff
@@ -171,26 +180,19 @@ void ResourceList::SetVisible(bool visible) {
...
-    
-    for (auto& editor : textureEditors) {
-        editor.second.SetVisible(false);
-    }
-    
-    for (auto& editor : soundEditors) {
-        editor.second.SetVisible(false);
-    }
+	
+	sceneEditor.SetVisible(false);
```
Comment (Chainsawkitten) [inline src/Editor/GUI/ResourceList.cpp]: Tabs

File: src/Editor/GUI/ResourceList.cpp [hunk 6]
```diff
@@ -151,14 +136,38 @@ void ResourceList::Show() {
...
-    // Sound editors.
-    for (Audio::SoundBuffer* sound : Hymn().sounds) {
-        if (soundEditors[sound].IsVisible()) {
-            soundEditors[sound].Show();
-        }
+
+
+    if (sceneEditor.entityPressed || texturePressed || modelPressed || soundPressed) {
+
+		sceneEditor.entityEditor.SetVisible(sceneEditor.entityPressed);
```
Comment (Chainsawkitten) [inline src/Editor/GUI/ResourceList.cpp]: Tabs

**Comment by Chainsawkitten** [comment]
Tabs everywhere. (Change your Visual Studio settings!)

File: src/Editor/GUI/Editors/SceneEditor.cpp [hunk 7]
```diff
@@ -8,19 +8,20 @@
 using namespace GUI;
 
 void SceneEditor::Show() {
-    if (ImGui::Begin(("Scene: " + *scene + "###Scene").c_str(), &visible)) {
+    if (ImGui::Begin(("Scene: " + Hymn().scenes[sceneIndex] + "###Scene").c_str(), &visible)) {
         ImGui::InputText("Name", name, 128);
-        *scene = name;
+		Hymn().scenes[sceneIndex] = name;
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.cpp]: Tabs

File: src/Editor/GUI/Editors/SceneEditor.cpp [hunk 8]
```diff
@@ -34,21 +35,18 @@ void SceneEditor::Show() {
...
     ImGui::End();
 }
 
-void SceneEditor::SetScene(std::string* scene) {
-    this->scene = scene;
-    
-    if (scene != nullptr)
-        strcpy(name, scene->c_str());
+void SceneEditor::SetScene(std::size_t sceneIndex) {
+	if (sceneIndex < Hymn().scenes.size()) {
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.cpp]: Tabs

File: src/Editor/GUI/Editors/SceneEditor.hpp [hunk 9]
```diff
@@ -33,13 +32,18 @@ namespace GUI {
             
             /// Save currently open scene to file.
             void Save() const;
+
+			///Did we press on an entity this frame.
```
Comment (Chainsawkitten) [inline src/Editor/GUI/Editors/SceneEditor.hpp]: Tabs

Comment (Sajdkick) [inline src/Editor/GUI/Editors/SceneEditor.hpp]: dammit

**Pull request merged**